### PR TITLE
ModelInstanceCollection picking

### PR DESCRIPTION
**Pull request by lilleyse**
This fixes picking for `ModelInstanceCollection`. This doesn't affect i3dm picking which continues to work fine.

The picked object that is returned is a `ModelInstance` object which contains the instance's modelMatrix, the underlying collection, the underlying model, and its instance id.

You can set the modelMatrix to move the instance dynamically.

@freder Is this the sort of functionality you had in mind?

### CODE REVIEW

**Comment by lilleyse** [comment]
One test is failing in CI: `culls when out of view and cull is true` but passes locally with `npm run test -- --browsers Electron --webgl-stub --failTaskOnError --suppressPassed`...

File: Source/Scene/ModelInstanceCollection.js [hunk 1]
```diff
@@ -54,6 +56,34 @@ define([
         FAILED : 3
     };
 
+    function ModelInstance(collection, modelMatrix, instanceId) {
```
Comment (pjcozzi) [inline Source/Scene/ModelInstanceCollection.js]: Should this be `@private` and in another file since it can be used outside of this file?

**Comment by pjcozzi** [comment]
> One test is failing in CI: culls when out of view and cull is true but passes locally with npm run test -- --browsers Electron --webgl-stub --failTaskOnError --suppressPassed...

Passes locally for me too.  Looks unrelated.  And I see different test failures in CI.
```
1) can create and destroy
     Widgets/CesiumInspector/CesiumInspector
     Expected HTMLNode to be HTMLNode.
    at Specs/Widgets/CesiumInspector/CesiumInspectorSpec.js:32:34
    at Object.<anonymous> (Specs/customizeJasmine.js:80:30)
2) can create and destroy
     Widgets/Geocoder/Geocoder
     Expected HTMLNode to be HTMLNode.
    at Specs/Widgets/Geocoder/GeocoderSpec.js:72:34
    at Object.<anonymous> (Specs/customizeJasmine.js:80:30)
     Expected 0 not to equal 0.
    at Specs/Widgets/Geocoder/GeocoderSpec.js:74:47
    at Object.<anonymous> (Specs/customizeJasmine.js:80:30)
3) mousedown event closes dropdown if target is not inside container
     Widgets/SceneModePicker/SceneModePicker
     TypeError: Cannot read property 'dispatchEvent' of null
    at fireMouseDown (Specs/DomEventSimulator.js:225:20)
    at Specs/Widgets/SceneModePicker/SceneModePickerSpec.js:54:13
    at Object.<anonymous> (Specs/customizeJasmine.js:80:30)
4) touchstart event closes dropdown if target is not inside container
     Widgets/SceneModePicker/SceneModePicker
     TypeError: Cannot read property 'dispatchEvent' of null
    at fireTouchStart (Specs/DomEventSimulator.js:246:20)
    at Specs/Widgets/SceneModePicker/SceneModePickerSpec.js:54:13
    at Object.<anonymous> (Specs/customizeJasmine.js:80:30)
```

**Comment by pjcozzi** [comment]
Code is also OK with me, just that one comment.

**Comment by freder** [comment]
@lilleyse 👍

**Comment by lilleyse** [comment]
Updated by putting `ModelInstance` into its own file and making sure to expand the bounding volume when an instance moves.

**Comment by pjcozzi** [comment]
Part of #4897

**Comment by pjcozzi** [comment]
Looks good!

**Comment by pjcozzi** [comment]
Thanks for the quick review, @freder!

**Pull request merged**